File: testdata/jennies/builders/array_append/JavaBuilder/sandbox/SomeStructBuilder.java
package sandbox;

import java.util.LinkedList;

public class SomeStructBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public SomeStructBuilder() {
        this.internal = new SomeStruct();
    }
    public SomeStructBuilder tags(String tags) {
		if (this.internal.tags == null) {
			this.internal.tags = new LinkedList<>();
		}
        this.internal.tags.add(tags);
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/constant_assignment/CSharpBuilder/src/Grafana/Foundation/Sandbox/SomeStructBuilder.cs
namespace Grafana.Foundation.Sandbox;


public class SomeStructBuilder : Cog.IBuilder<SomeStruct>
{
    protected readonly SomeStruct @internal;

    public SomeStructBuilder()
    {
        this.@internal = new SomeStruct();
    }

    public SomeStructBuilder Editable()
    {
        this.@internal.Editable = true;
        return this;
    }

    public SomeStructBuilder ReadonlyArg()
    {
        this.@internal.Editable = false;
        return this;
    }

    public SomeStructBuilder AutoRefresh()
    {
        this.@internal.AutoRefresh = true;
        return this;
    }

    public SomeStructBuilder NoAutoRefresh()
    {
        this.@internal.AutoRefresh = false;
        return this;
    }

    public SomeStruct Build()
    {
        return this.@internal;
    }
}
